### PR TITLE
Added compatibility to Spring Boot v2.5.0

### PR DESCRIPTION
**Pull request by canarddemagret**
It's the same PR than [lhmoraes](https://github.com/lhmoraes), but with codestyle and tests done.

### CODE REVIEW

**Comment by ArtemchukAndrei94** [comment]
Hello. I want you to add compatibility with spring boot 2.5.0. Please check this pull request and if everything is ok, merge with main. Thanks.

File: spring-multirabbit/src/main/java/org/springframework/boot/autoconfigure/amqp/MultiRabbitAutoConfiguration.java [hunk 1]
```diff
@@ -128,10 +128,12 @@ public ConnectionFactory routingConnectionFactory(
                 final ResourceLoader resourceLoader,
                 final ObjectProvider<CredentialsProvider> credentialsProvider,
                 final ObjectProvider<CredentialsRefreshService> credentialsRefreshService,
-                final ObjectProvider<ConnectionNameStrategy> connectionNameStrategy) throws Exception {
+                final ObjectProvider<ConnectionNameStrategy> connectionNameStrategy,
+                final ObjectProvider<ConnectionFactoryCustomizer> connectionFactoryCustomizers) throws Exception {
```
Comment (rwanderc) [inline spring-multirabbit/src/main/java/org/springframework/boot/autoconfigure/amqp/MultiRabbitAutoConfiguration.java:132]: I think it should be `connectionFactoryCustomizer` without `s`, right?

File: spring-multirabbit/src/test/java/org/springframework/boot/autoconfigure/amqp/MultiRabbitConnectionFactoryCreatorTest.java [hunk 2]
```diff
@@ -54,6 +54,9 @@ class MultiRabbitConnectionFactoryCreatorTest {
     @Mock
     private ObjectProvider<CredentialsRefreshService> credentialsRefreshService;
 
+    @Mock
+    private ObjectProvider<ConnectionFactoryCustomizer> connectionFactoryCustomizers;
```
Comment (rwanderc) [inline spring-multirabbit/src/test/java/org/springframework/boot/autoconfigure/amqp/MultiRabbitConnectionFactoryCreatorTest.java:58]: Also the plural comment here.

File: spring-multirabbit/src/test/java/org/springframework/boot/autoconfigure/amqp/MultiRabbitConnectionFactoryCreatorTest.java [hunk 3]
```diff
@@ -100,11 +103,12 @@ void shouldInstantiateRoutingConnectionFactory() throws Exception {
         externalWrapper.addConnectionFactory(DUMMY_KEY, connectionFactory1, containerFactory, rabbitAdmin);
 
         when(springFactoryCreator.rabbitConnectionFactory(rabbitProperties, resourceLoader, credentialsProvider,
-                credentialsRefreshService, connectionNameStrategy)).thenReturn(new CachingConnectionFactory());
+                credentialsRefreshService, connectionNameStrategy,
+                connectionFactoryCustomizers)).thenReturn(new CachingConnectionFactory());
```
Comment (rwanderc) [inline spring-multirabbit/src/test/java/org/springframework/boot/autoconfigure/amqp/MultiRabbitConnectionFactoryCreatorTest.java:107]: you can use the previous line as much as possible, and leave only .thenReturn(...); at the new line.

File: spring-multirabbit/src/test/java/org/springframework/boot/autoconfigure/amqp/MultiRabbitConnectionFactoryCreatorTest.java [hunk 4]
```diff
@@ -196,65 +202,70 @@ void shouldInstantiateMultiRabbitConnectionFactoryWrapperWithDefaultConnection()
         final MultiRabbitConnectionFactoryWrapper externalWrapper = new MultiRabbitConnectionFactoryWrapper();
 
         when(springFactoryCreator.rabbitConnectionFactory(rabbitProperties, resourceLoader, credentialsProvider,
-                credentialsRefreshService, connectionNameStrategy)).thenReturn(new CachingConnectionFactory());
+                credentialsRefreshService, connectionNameStrategy,
+                connectionFactoryCustomizers)).thenReturn(new CachingConnectionFactory());
```
Comment (rwanderc) [inline spring-multirabbit/src/test/java/org/springframework/boot/autoconfigure/amqp/MultiRabbitConnectionFactoryCreatorTest.java:206]: you can use the previous line as much as possible, and leave only .thenReturn(...); at the new line.

**Comment by rwanderc** [comment]
Thank you for your contribution @canarddemagret.
The PR looks good, just a few formatting comments.
Please also update the SpringBoot version of the examples, so it's aligned with the project and fix the build.

**Review by rwanderc — APPROVED** [review]

**Review by rwanderc — CHANGES_REQUESTED** [review]
Thank you for your contribution @canarddemagret.
The PR looks good, just a few formatting comments.
Please also update the SpringBoot version of the examples, so it's aligned with the project and fix the build.

**Comment by ArtemchukAndrei94** [comment]
Hello. I created it myself pr
https://github.com/freenowtech/spring-multirabbit/pull/84, check review
please. Thanks.

вс, 17 апр. 2022 г. в 17:55, Wander Costa ***@***.***>:

**Comment by ArtemchukAndrei94** [comment]
I suggest waiting a few days for canarddemagret, and if he is not active, I suggest returning my pr and looking at it and if canarddemagret contacts and makes corrections, that will be fine too. Thanks.

**Comment by canarddemagret** [comment]
I am currently in holiday, I will be able to finish the PR friday

**Comment by rwanderc** [comment]
> I suggest waiting a few days for canarddemagret, and if he is not active, I suggest returning my pr and looking at it and if canarddemagret contacts and makes corrections, that will be fine too. Thanks.

@canarddemagret's PR was missing only a few minor changes. I'm proceeding with the fixes on @canarddemagret PR so as to push this to the main branch and eventually release the 2.5.0.

**Review by rwanderc — APPROVED** [review]

**Comment by ArtemchukAndrei94** [comment]
Thank you so much!

**Pull request merged**